dockers/docker-dhcp-relay/docker-dhcp-relay.supervisord.conf.j2

```diff
@@ -43,7 +43,7 @@ dependent_startup_wait_for=rsyslogd:running
 {# Count how many VLANs require a DHCP relay agent... #}
 {% set num_relays = { 'count': 0 } %}
 {% for vlan_name in VLAN %}
-{% if VLAN[vlan_name]['dhcp_servers'] %}
+{% if VLAN[vlan_name]['dhcp_servers'] and vlan_name in VLAN_INTERFACE %}
```

Comment:
Should we instead change this to:

```python
{% if VLAN_INTERFACE %}
{# Count how many VLANs require a DHCP relay agent... #}
{% set num_relays = { 'count': 0 } %}
{% for vlan_name in VLAN_INTERFACE %}
{% if VLAN and vlan_name in VLAN and VLAN[vlan_name]['dhcp_servers'] %}
{% set _dummy = num_relays.update({'count': num_relays.count + 1}) %}
{% endif %}
{% endfor %}
```

Reply:
The `VLAN_INTERFACE` has other entries that are not vlan maps. here is an example:
```
    "VLAN_INTERFACE": {
        "Vlan701": {
            "proxy_arp": "enabled",
            "grat_arp": "enabled"
        },
        "Vlan701|2603:10e2:3b0:8000::1/64": {},
        "Vlan701|10.138.12.1/26": {}
    },
```

Reply:
Ah, right. Technically, I think my proposal would still work, but `{% for vlan_name in VLAN_INTERFACE %}` would not be technically correct, because `vlan_name` would not be guaranteed to be an actual VLAN name.

Reply:
I see, the `vlan_name in VLAN` should guard against this.

Reply:
Right. But I do think it feels a bit "wrong", as I described above. Thoughts?

Reply:
I guess it boils down to nomenclature issue. The other way is to check for `|` and then do splitting to extract the vlan. The presence redis kind keys is strange here where we have `json` luxury.

Reply:
Yeah, great point. Why don't we simply store the IPv4 and IPv6 IP addresses in lists in the one VLAN_INTERFACE entry? @prsunny : Is there a reason?

Reply:
AFAIK, its a legacy design. One advantage is, it helps in deletion by having it as single key.